core/issuers.chainmail

```diff
@@ -1,56 +1,152 @@
-class Assay[Quantity] {
-  type Label = { issuer: Issuer, description };
-  type Amount = { label: Label, quantity: Quantity };
-  
-  getLabel() ::Label;
-  make(allegedQuantity ?Quantity) ::Amount;
-  vouch(amount ?Amount) ::Amount
-  coerce(amountLike ?Amount) ::Amount;
-  quantity(amount ?Amount) ::Quantity;
-  empty() ::Amount;
-  isEmpty(amount ?Amount) ::boolean;
-  includes(leftAmount ?Amount, rightAmount ?Amount) ::boolean;
-  with(leftAmount ?Amount, rightAmount ?Amount) ::Amount;
-  without(leftAmount ?Amount, rightAmount ?Amount) ::Amount;
+/**
+ * Holding an Issuer provides the ability to create amounts and empty purses,
+ * but confers no rights. It is also the mechanism used to get exclusive access
+ * to a Purse or Payment that you already hold, or to burn some or all of the
+ * contained rights.
+ */
+interface Issuer (Amount (Quantity)) {
+  /**
+   * Get the label for this Issuer.
+   *
+   * Labels can be used to manually construct amounts
+   */
+  getLabel() -> (Comparable);
+
+  /* Get the Assay for this Issuer. */
+  getAssay() -> (Assay);
+
+  /** Make an Amount that contains the indicated quantity. */
+  makeAmount(quantity :Quantity) -> (Amount);
+
+  /** Make an empty purse associated with this kind of right. */
+  makeEmptyPurse(name :String) -> (Purse);
+
+  /**
+   * Make a new Payment that has exclusive rights to the amount, only if src had
+   * those rights, exclusively or not. The src payment retains any excess not
+   * included in amount.
+   */
+  getExclusive(amount :Amount, src :Payment, name :String = 'a payment')
+    -> (Payment);
+
+  /**
+   * Make a new Payment that has exclusive rights to all the contents of src.
+   */
+  getExclusiveAll(src :Payment, name :String = 'a payment') -> (Payment);
+
+  /**
+   * Burn amount of the rights from src. If the rights are fungible, this
```

Comment:
I think there's no fungible/non-fungible distinction here. In both cases, the rights are deposited into a purse (and therefore taken from the payment). The purse is dropped, therefore ensuring that no one has access to the rights, even the holder of the payment originally.

Reply:
The fact that it's implemented by dropping a purse isn't part of the visible specification. And that approach isn't how 'burning' is done with cryptocurrencies, as I understand it.

I explain the distinction because as far as I can tell, the 'burn' metaphor only describes what happens to non-fungible values. With BitCoin, I can burn a particular UTXO, but burn doesn't apply to "1 bitcoin of value". I can figure out how to burn an invite, or a right to a particular pixel from the (blockchain) meaning of burn. It doesn't seem to me that the metaphor says what happens to fungible rights.

Reply:
In Ethereum, you burn a fungible amount by sending it to an invalid address. It's true that they don't have purses, but a purse that is dropped and accessible by no one is very close to an account that is accessible by no one. 

I found this definition of "burn" from a [slightly questionable source](https://coinsutra.com/proof-of-burn/). I think it's sound and a widely used definition: 

> Coin burn in cryptocurrencies means sending some of the coins of native cryptocurrency or some other currency to a public address from which those particular coins can never be spent because the private keys of such an address are unobtainable.

Our burn method is doing the same thing except we are using purses rather than accounts. And our burn method acts exactly the same regardless of whether the tokens are fungible or non-fungible. I think this discussion may have gotten mixed up in the discussion of `destroy` for the pixel demo, which does depend on an identifiable pixel to be destroyed and is very different than `burn` in its mechanism.

Reply:
I dropped the extraneous explanation.